### PR TITLE
[PUPIL-759] - Cycle 1 support

### PR DESCRIPTION
**Pull request by BTitterington**
Richard knows how to test the actually functionality, and has done the majority of that already.

**Devs**: Apologies this is a big one to review, it was a bit unavoidable with the way things evolved.

Due to it's size I think it's acceptable to scan for any obvious mistakes or anti-patterns, but I don't think it's required to pickup on any complex mistakes, as that would be very difficult in this instance. I think Richard has the functionality covered also.

## Description

This PR makes cycle 1 data available in the pupil browse experience. More info about the implementation, testing and AC can be found on the [ticket](https://www.notion.so/oaknationalacademy/Cycle-1-Support-Stage-1-4a044f919bda408782919481aab06fc5)

## How to test

1. Go to https://deploy-preview-2479--oak-web-application.netlify.thenational.academy/pupils/years

See [ticket](https://www.notion.so/oaknationalacademy/Cycle-1-Support-Stage-1-4a044f919bda408782919481aab06fc5) for details

## Checklist

- [x] Added or updated tests where appropriate
- [x] Manually tested across browsers / devices
- [x] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [x] Does this PR update a package with a breaking change

### CODE REVIEW

**Comment by BTitterington** [comment]
@RBrownsett - For info - I've updated so that lessons now work again. I also think this branch is now pretty much stable/done, I just need to resolve the security hotspot issue, which shouldn't effect functionality 🤞.

**Comment by RBrownsett** [comment]
@BTitterington looking good - did we say that for combined science year 10/11 we would show all (new and old)? At the moment it shows only the new units? Same for Biology and Chemistry.

**Comment by BTitterington** [comment]
@RBrownsett Yeah, I did just realise that after posting also :) 

I'm looking at it now to see if it's simple. So code may change a bit more sorry.

**Review by RBrownsett — COMMENTED** [review]
Happy from initial testing that this can go to dev review - full testing details have been added to the ticket. Further tickets has been raised to reduce scope creep (PUPIL-766/PUPIL-780/PUPIL-779)

**Comment by messmike** [comment]
Small bug on the tier programme page where the tooltip appears with a gap when opened:
Screenshot 2024-06-12 at 11 12 55 am

**Review by messmike — APPROVED** [review]
Apart from the minor bug mentioned in the ticket, this is looking good from my perspective

File: src/components/PupilComponents/BrowseTierSelector/BrowseTierSelector.tsx [hunk 1]
```diff
@@ -9,36 +9,28 @@ import { resolveOakHref } from "@/common-lib/urls";
...
-  });
+  const orderedTiers = tiers.sort(
+    (a, b) => (a.tierDisplayOrder ?? 0) - (b.tierDisplayOrder ?? 0),
+  );
 
   const programmeSlugs = orderedTiers.map(
     (tier) =>
       `${baseSlug}-${tier.tierSlug}${examboardSlug ? `-${examboardSlug}` : ""}${
-        isLegacy ? "-l" : ""
+        tier.isLegacy ? "-l" : ""
```
Comment (njblore) [inline src/components/PupilComponents/BrowseTierSelector/BrowseTierSelector.tsx:33]: we have a collection of slugModifiers in utils which includes a few functions for constructing / validating legacy URLs, might be worth looking to see if any are useful

Comment (BTitterington) [reply src/components/PupilComponents/BrowseTierSelector/BrowseTierSelector.tsx:33]: @njblore Couldn't see one that would help for this, I think isSlugLegacy could be used in one spot though. Thanks

**Review by RBrownsett — APPROVED** [review]
Approved! Good work! All testing details added to the ticket

**Pull request merged**